Errors with development mode - missing await in pydle.features.account?

When running pydle with [Python development mode](https://docs.python.org/3/library/devmode.html) enabled, errors like this appear in the program output.
```
 asyncio [ERROR] <CoroWrapper AccountSupport.whois() running at /home/bridge/venv/lib/python3.8/site-packages/pydle/features/account.py:25, created at /home/bridge/venv/lib/python3.8/site-packages/pydle/features/account.py:22> was never yielded from
 Coroutine object created at (most recent call last, truncated to 10 last lines):
   File "/home/bridge/venv/lib/python3.8/site-packages/aiorun.py", line 267, in run
     loop.run_forever()
   File "/usr/lib/python3.8/asyncio/base_events.py", line 570, in run_forever
     self._run_once()
   File "/usr/lib/python3.8/asyncio/base_events.py", line 1851, in _run_once
     handle._run()
   File "/usr/lib/python3.8/asyncio/events.py", line 81, in _run
     self._context.run(self._callback, *self._args)
   File "/home/bridge/venv/lib/python3.8/site-packages/pydle/features/ircv3/ircv3_2.py", line 82, in on_raw
     await super().on_raw(message)
   File "/home/bridge/venv/lib/python3.8/site-packages/pydle/client.py", line 422, in on_raw
     await handler(message)
   File "/home/bridge/venv/lib/python3.8/site-packages/pydle/features/rfc1459/client.py", line 660, in on_raw_nick
     self._rename_user(nick, new)
   File "/home/bridge/venv/lib/python3.8/site-packages/pydle/features/ircv3/ircv3_1.py", line 22, in _rename_user
     super()._rename_user(user, new)
   File "/home/bridge/venv/lib/python3.8/site-packages/pydle/features/account.py", line 22, in _rename_user
     self.whois(new)
  ```
`self.whois` is async, and it doesn't `await` it.. but it can't just do that because _rename_user isn't async.
I'm not sure about the best way to fix this so I'm just filing an issue.
This occurs with pydle 0.9.4.

--- a/pydle/client.py
+++ b/pydle/client.py
@@ -197,21 +197,21 @@ class BasicClient:
             'hostname': None
         }
 
-    def _sync_user(self, nick, metadata):
+    async def _sync_user(self, nick, metadata):
         # Create user in database.
         if nick not in self.users:
-            self._create_user(nick)
+            await self._create_user(nick)
             if nick not in self.users:
                 return
         self.users[nick].update(metadata)
 
-    def _rename_user(self, user, new):
+    async def _rename_user(self, user, new):
         if user in self.users:
             self.users[new] = self.users[user]
             self.users[new]['nickname'] = new
             del self.users[user]
         else:
-            self._create_user(new)
+            await self._create_user(new)
             if new not in self.users:
                 return
 

--- a/pydle/features/account.py
+++ b/pydle/features/account.py
@@ -16,7 +16,7 @@ class AccountSupport(rfc1459.RFC1459Support):
             })
 
     async def _rename_user(self, user, new):
-        super()._rename_user(user, new)
+        await super()._rename_user(user, new)
         # Unset account info to be certain until we get a new response.
         await self._sync_user(new, {'account': None, 'identified': False})
         await self.whois(new)
@@ -24,6 +24,7 @@ class AccountSupport(rfc1459.RFC1459Support):
     ## IRC API.
     async def whois(self, nickname):
         info = await super().whois(nickname)
+        if info is None: return info
         info.setdefault('account', None)
         info.setdefault('identified', False)
         return info

--- a/pydle/features/account.py
+++ b/pydle/features/account.py
@@ -24,7 +24,8 @@ class AccountSupport(rfc1459.RFC1459Support):
     ## IRC API.
     async def whois(self, nickname):
         info = await super().whois(nickname)
-        if info is None: return info
+        if info is None:
+            return info
         info.setdefault('account', None)
         info.setdefault('identified', False)
         return info

--- a/pydle/features/account.py
+++ b/pydle/features/account.py
@@ -15,11 +15,11 @@ class AccountSupport(rfc1459.RFC1459Support):
                 'identified': False
             })
 
-    def _rename_user(self, user, new):
+    async def _rename_user(self, user, new):
         super()._rename_user(user, new)
         # Unset account info to be certain until we get a new response.
-        self._sync_user(new, {'account': None, 'identified': False})
-        self.whois(new)
+        await self._sync_user(new, {'account': None, 'identified': False})
+        await self.whois(new)
 
     ## IRC API.
     async def whois(self, nickname):
@@ -38,7 +38,7 @@ class AccountSupport(rfc1459.RFC1459Support):
         }
 
         if nickname in self.users:
-            self._sync_user(nickname, info)
+            await self._sync_user(nickname, info)
         if nickname in self._pending['whois']:
             self._whois_info[nickname].update(info)
 
@@ -51,6 +51,6 @@ class AccountSupport(rfc1459.RFC1459Support):
         }
 
         if nickname in self.users:
-            self._sync_user(nickname, info)
+            await self._sync_user(nickname, info)
         if nickname in self._pending['whois']:
             self._whois_info[nickname].update(info)

--- a/pydle/features/ircv3/ircv3_1.py
+++ b/pydle/features/ircv3/ircv3_1.py
@@ -12,7 +12,7 @@ NO_ACCOUNT = '*'
 class IRCv3_1Support(sasl.SASLSupport, cap.CapabilityNegotiationSupport, account.AccountSupport, tls.TLSSupport):
     """ Support for IRCv3.1's base and optional extensions. """
 
-    def _rename_user(self, user, new):
+    async def _rename_user(self, user, new):
         # If the server supports account-notify, we will be told about the registration status changing.
         # As such, we can skip the song and dance pydle.features.account does.
         if self._capabilities.get('account-notify', False):
@@ -22,7 +22,7 @@ class IRCv3_1Support(sasl.SASLSupport, cap.CapabilityNegotiationSupport, account
         super()._rename_user(user, new)
 
         if self._capabilities.get('account-notify', False):
-            self._sync_user(new, {'account': account, 'identified': identified})
+            await self._sync_user(new, {'account': account, 'identified': identified})
 
     ## IRC callbacks.
 
@@ -85,7 +85,7 @@ class IRCv3_1Support(sasl.SASLSupport, cap.CapabilityNegotiationSupport, account
             nick, metadata = self._parse_user(message.source)
             channels, account, realname = message.params
 
-            self._sync_user(nick, metadata)
+            await self._sync_user(nick, metadata)
 
             # Emit a fake join message.
             fakemsg = self._create_message('JOIN', channels, source=message.source)

--- a/pydle/features/ircv3/ircv3_1.py
+++ b/pydle/features/ircv3/ircv3_1.py
@@ -60,11 +60,11 @@ class IRCv3_1Support(sasl.SASLSupport, cap.CapabilityNegotiationSupport, account
         if nick not in self.users:
             return
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
         if account == NO_ACCOUNT:
             self._sync_user(nick, { 'account': None, 'identified': False })
         else:
-            self._sync_user(nick, { 'account': account, 'identified': True })
+            await self._sync_user(nick, { 'account': account, 'identified': True })
 
     async def on_raw_away(self, message):
         """ Process AWAY messages. """
@@ -75,7 +75,7 @@ class IRCv3_1Support(sasl.SASLSupport, cap.CapabilityNegotiationSupport, account
         if nick not in self.users:
             return
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
         self.users[nick]['away'] = len(message.params) > 0
         self.users[nick]['away_message'] = message.params[0] if len(message.params) > 0 else None
 

--- a/pydle/features/ircv3/ircv3_1.py
+++ b/pydle/features/ircv3/ircv3_1.py
@@ -62,7 +62,7 @@ class IRCv3_1Support(sasl.SASLSupport, cap.CapabilityNegotiationSupport, account
 
         await self._sync_user(nick, metadata)
         if account == NO_ACCOUNT:
-            self._sync_user(nick, { 'account': None, 'identified': False })
+            await self._sync_user(nick, { 'account': None, 'identified': False })
         else:
             await self._sync_user(nick, { 'account': account, 'identified': True })
 

--- a/pydle/features/ircv3/ircv3_1.py
+++ b/pydle/features/ircv3/ircv3_1.py
@@ -19,7 +19,7 @@ class IRCv3_1Support(sasl.SASLSupport, cap.CapabilityNegotiationSupport, account
             account = self.users.get(user, {}).get('account', None)
             identified = self.users.get(user, {}).get('identified', False)
 
-        super()._rename_user(user, new)
+        await super()._rename_user(user, new)
 
         if self._capabilities.get('account-notify', False):
             await self._sync_user(new, {'account': account, 'identified': identified})

--- a/pydle/features/ircv3/ircv3_2.py
+++ b/pydle/features/ircv3/ircv3_2.py
@@ -78,7 +78,7 @@ class IRCv3_2Support(metadata.MetadataSupport, monitor.MonitoringSupport, tags.T
                     'identified': True,
                     'account': message.tags['account']
                 }
-                self._sync_user(nick, metadata)
+                await self._sync_user(nick, metadata)
         await super().on_raw(message)
 
     async def on_raw_chghost(self, message):
@@ -95,4 +95,4 @@ class IRCv3_2Support(metadata.MetadataSupport, monitor.MonitoringSupport, tags.T
             'username': message.params[0],
             'hostname': message.params[1]
         }
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)

--- a/pydle/features/ircv3/metadata.py
+++ b/pydle/features/ircv3/metadata.py
@@ -61,7 +61,7 @@ class MetadataSupport(cap.CapabilityNegotiationSupport):
             visibility = None
 
         if target in self.users:
-            self._sync_user(target, targetmeta)
+            await self._sync_user(target, targetmeta)
         await self.on_metadata(target, key, value, visibility=visibility)
 
     async def on_raw_760(self, message):
@@ -72,7 +72,7 @@ class MetadataSupport(cap.CapabilityNegotiationSupport):
         if target not in self._pending['whois']:
             return
         if target in self.users:
-            self._sync_user(target, targetmeta)
+            await self._sync_user(target, targetmeta)
 
         self._whois_info[target].setdefault('metadata', {})
         self._whois_info[target]['metadata'][key] = value
@@ -86,7 +86,7 @@ class MetadataSupport(cap.CapabilityNegotiationSupport):
         if target not in self._pending['metadata']:
             return
         if target in self.users:
-            self._sync_user(target, targetmeta)
+            await self._sync_user(target, targetmeta)
 
         self._metadata_info[target][key] = value
 
@@ -112,7 +112,7 @@ class MetadataSupport(cap.CapabilityNegotiationSupport):
         if target not in self._pending['metadata']:
             return
         if target in self.users:
-            self._sync_user(target, targetmeta)
+            await self._sync_user(target, targetmeta)
 
         self._metadata_queue.remove(target)
         del self._metadata_info[target]

--- a/pydle/features/ircv3/monitor.py
+++ b/pydle/features/ircv3/monitor.py
@@ -77,7 +77,7 @@ class MonitoringSupport(isupport.ISUPPORTSupport):
         """ Someone we are monitoring just came online. """
         for target in message.params[1].split(','):
             nickname, metadata = self._parse_user(target)
-            self._sync_user(nickname, metadata)
+            await self._sync_user(nickname, metadata)
             await self.on_user_online(nickname)
 
     async def on_raw_731(self, message):

--- a/pydle/features/rfc1459/client.py
+++ b/pydle/features/rfc1459/client.py
@@ -657,7 +657,7 @@ class RFC1459Support(BasicClient):
             self.nickname = new
 
         # Go through all user lists and replace.
-        self._rename_user(nick, new)
+        await self._rename_user(nick, new)
 
         # Call handler.
         await self.on_nick_change(nick, new)

--- a/pydle/features/rfc1459/client.py
+++ b/pydle/features/rfc1459/client.py
@@ -84,8 +84,8 @@ class RFC1459Support(BasicClient):
                 'away_message': None,
             })
 
-    def _rename_user(self, user, new):
-        super()._rename_user(user, new)
+    async def _rename_user(self, user, new):
+        await super()._rename_user(user, new)
 
         # Rename in mode lists, too.
         for ch in self.channels.values():
@@ -560,7 +560,7 @@ class RFC1459Support(BasicClient):
     async def on_raw_join(self, message):
         """ JOIN command. """
         nick, metadata = self._parse_user(message.source)
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
 
         channels = message.params[0].split(',')
         if self.is_same_nick(self.nickname, nick):
@@ -628,7 +628,7 @@ class RFC1459Support(BasicClient):
         nick, metadata = self._parse_user(message.source)
         target, modes = message.params[0], message.params[1:]
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
         if self.is_channel(target):
             if self.in_channel(target):
                 # Parse modes.
@@ -637,7 +637,7 @@ class RFC1459Support(BasicClient):
                 await self.on_mode_change(target, modes, nick)
         else:
             target, targetmeta = self._parse_user(target)
-            self._sync_user(target, targetmeta)
+            await self._sync_user(target, targetmeta)
 
             # Update own modes.
             if self.is_same_nick(self.nickname, nick):
@@ -650,7 +650,7 @@ class RFC1459Support(BasicClient):
         nick, metadata = self._parse_user(message.source)
         new = message.params[0]
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
         # Acknowledgement of nickname change: set it internally, too.
         # Alternatively, we were force nick-changed. Nothing much we can do about it.
         if self.is_same_nick(self.nickname, nick):
@@ -667,7 +667,7 @@ class RFC1459Support(BasicClient):
         nick, metadata = self._parse_user(message.source)
         target, message = message.params
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
 
         await self.on_notice(target, nick, message)
         if self.is_channel(target):
@@ -684,7 +684,7 @@ class RFC1459Support(BasicClient):
         else:
             reason = None
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
         if self.is_same_nick(self.nickname, nick):
             # We left the channel. Remove from channel list. :(
             for channel in channels:
@@ -707,7 +707,7 @@ class RFC1459Support(BasicClient):
         nick, metadata = self._parse_user(message.source)
         target, message = message.params
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
 
         await self.on_message(target, nick, message)
         if self.is_channel(target):
@@ -798,7 +798,7 @@ class RFC1459Support(BasicClient):
             'realname': realname
         }
 
-        self._sync_user(nickname, info)
+        await self._sync_user(nickname, info)
         if nickname in self._pending['whois']:
             self._whois_info[nickname].update(info)
 
@@ -926,7 +926,7 @@ class RFC1459Support(BasicClient):
             if not nick:
                 # nonsense nickname
                 continue
-            self._sync_user(nick, metadata)
+            await self._sync_user(nick, metadata)
 
             # Get prefixes.
             prefixes = set(entry.replace(safe_entry, ''))

--- a/pydle/features/rfc1459/client.py
+++ b/pydle/features/rfc1459/client.py
@@ -719,7 +719,7 @@ class RFC1459Support(BasicClient):
         """ QUIT command. """
         nick, metadata = self._parse_user(message.source)
 
-        self._sync_user(nick, metadata)
+        await self._sync_user(nick, metadata)
         if message.params:
             reason = message.params[0]
         else:

--- a/pydle/features/rpl_whoishost/rpl_whoishost.py
+++ b/pydle/features/rpl_whoishost/rpl_whoishost.py
@@ -20,6 +20,7 @@ class RplWhoisHostSupport(RFC1459Support):
 
     async def whois(self, nickname):
         info = await super().whois(nickname)
+        if info is None: return info
         info.setdefault("real_ip_address", None)
         info.setdefault("real_hostname", None)
         return info

--- a/pydle/features/rpl_whoishost/rpl_whoishost.py
+++ b/pydle/features/rpl_whoishost/rpl_whoishost.py
@@ -20,7 +20,8 @@ class RplWhoisHostSupport(RFC1459Support):
 
     async def whois(self, nickname):
         info = await super().whois(nickname)
-        if info is None: return info
+        if info is None:
+            return info
         info.setdefault("real_ip_address", None)
         info.setdefault("real_hostname", None)
         return info

--- a/pydle/features/rpl_whoishost/rpl_whoishost.py
+++ b/pydle/features/rpl_whoishost/rpl_whoishost.py
@@ -13,7 +13,7 @@ class RplWhoisHostSupport(RFC1459Support):
         host = data[-2]
 
         meta = {"real_ip_address": ip_addr, "real_hostname": host}
-        self._sync_user(target, meta)
+        await self._sync_user(target, meta)
         if target in self._whois_info:
             self._whois_info[target]["real_ip_address"] = ip_addr
             self._whois_info[target]["real_hostname"] = host

--- a/pydle/features/tls.py
+++ b/pydle/features/tls.py
@@ -58,6 +58,7 @@ class TLSSupport(rfc1459.RFC1459Support):
 
     async def whois(self, nickname):
         info = await super().whois(nickname)
+        if info is None: return info
         info.setdefault('secure', False)
         return info
 

--- a/pydle/features/tls.py
+++ b/pydle/features/tls.py
@@ -58,7 +58,8 @@ class TLSSupport(rfc1459.RFC1459Support):
 
     async def whois(self, nickname):
         info = await super().whois(nickname)
-        if info is None: return info
+        if info is None:
+            return info
         info.setdefault('secure', False)
         return info
 

--- a/pydle/features/whox.py
+++ b/pydle/features/whox.py
@@ -25,10 +25,10 @@ class WHOXSupport(isupport.ISUPPORTSupport, account.AccountSupport):
             # Find account name of person.
             pass
     
-    def _create_user(self, nickname):
+    async def _create_user(self, nickname):
         super()._create_user(nickname)
         if self.registered and 'WHOX' not in self._isupport:
-            self.whois(nickname)
+            await self.whois(nickname)
 
     async def on_raw_354(self, message):
         """ WHOX results have arrived. """
@@ -48,4 +48,4 @@ class WHOXSupport(isupport.ISUPPORTSupport, account.AccountSupport):
             metadata['identified'] = True
             metadata['account'] = message.params[5]
 
-        self._sync_user(metadata['nickname'], metadata)
+        await self._sync_user(metadata['nickname'], metadata)

